src/Microsoft.OData.Client/Metadata/ClientTypeUtil.cs

```diff
@@ -765,5 +776,59 @@ private static bool IsOverride(Type type, PropertyInfo propertyInfo)
 
             return false;
         }
+
+        /// <summary>
+        /// Returns true if the <paramref name="instance"/> is an instance of an open type in the <paramref name="model"/>.
+        /// </summary>
+        /// <param name="instance">The instance to examine</param>
+        /// <param name="model">The client model.</param>
+        /// <returns></returns>
+        internal static bool IsInstanceOfOpenType(object instance, ClientEdmModel model)
+        {
+            Debug.Assert(instance != null, $"{nameof(instance)} !=null");
+            Debug.Assert(model != null, $"{nameof(model)} !=null");
+
+            Type clientType = instance.GetType();
+            ClientTypeAnnotation clientTypeAnnotation = model.GetClientTypeAnnotation(clientType);
+
+            Debug.Assert(clientTypeAnnotation != null, $"{nameof(clientTypeAnnotation)} != null");
+
+            // TODO: Populate dynamic properties only if this is an open (entity or complex) type
+            return clientTypeAnnotation.EdmType.IsOpen();
+        }
+
+        /// <summary>
+        /// Returns true if the <paramref name="instance"/> contains an non-null dictionary property of string and object
+        /// The dictionary should also not be decorated with IgnoreClientPropertyAttribute
+        /// </summary>
+        /// <param name="instance">Object with expected dictionary property</param>
+        /// <param name="dynamicPropertiesDictionary">Reference to the dictionary</param>
+        /// <returns>true if expected dictionary is found</returns>
+        internal static bool TryGetDynamicPropertiesDictionary(object instance, out IDictionary<string, object> dynamicPropertiesDictionary)
+        {
+            Debug.Assert(instance != null, $"{nameof(instance)} != null");
+
+            dynamicPropertiesDictionary = default(IDictionary<string, object>);
+
+            PropertyInfo propertyInfo = instance.GetType().GetProperties().Where(p =>
+                                !p.GetCustomAttributes(typeof(IgnoreClientPropertyAttribute), true).Any() &&
+                                typeof(IDictionary<string, object>).IsAssignableFrom(p.PropertyType)
+                            ).FirstOrDefault();
+
+            if (propertyInfo == null)
+            {
+                return false;
+            }
+
+            dynamicPropertiesDictionary = (IDictionary<string, object>)propertyInfo.GetValue(instance);
+
+            // Is property initialized?
+            if (dynamicPropertiesDictionary == null)
+            {
+                return false;

```

Comment:
>return false; [](start = 16, length = 13)

if it's not initialized, can you create the dict and return it?

Reply:
@xuzhg I'll look into this. Initializing the dictionary though would only be relevant during materialization. In a serialization scenario, the developer would have had to initialize the property in order to add the dynamic properties

Reply:
@xuzhg Addressed your review feedback